internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -56,7 +56,7 @@ func (i *Infra) createOrUpdateConfigMap(ctx context.Context, r ResourceRender) e
 	}
 
 	return i.Client.CreateOrUpdate(ctx, key, current, cm, func() bool {
-		return !reflect.DeepEqual(cm.Data, current.Data)
+		return !reflect.DeepEqual(cm.Data, current.Data) || !reflect.DeepEqual(cm.BinaryData, current.BinaryData)
```

Comment:
why do we need to compare `BinaryData` here ?

Reply:
Users can set the cm with binary data